Source/Assimil8or/Assimil8or.cpp

```diff
@@ -667,375 +894,21 @@ void Assimil8orPreset::parse (juce::StringArray presetLines)
         }
 
         LogParsing (juce::String (scopeDepth) + "-" + presetLine.trimStart ());
-        const auto key { presetLine.upToFirstOccurrenceOf (":", false, false).trim () };
-        const auto value { presetLine.fromFirstOccurrenceOf (":", false, false).trim () };
-
-        auto keyIs = [&key] (const juce::String& desiredKey)
+        key = presetLine.upToFirstOccurrenceOf (":", false, false).trim ();
+        key = key.upToFirstOccurrenceOf(" ", false, false);
```

Comment:
I understand why you've added this, but.. it will have broken things. The 'key' (probably the wrong name for it) is usually a string that is the parameter (Name, LoopMode, etc), but for the sections it is `Preset 1`, `Channel 3`, `Zone 2`, etc. That number is retrieved by the `getParameterIndex`. With this change, the number is now missing from the string, and it will always come up as 0. I think 0 is invalid, so we could even add an assert check for that in `getParameterIndex`. Instead of stripping it off here, make `keyIs` a member function and use it on the key when calling `find` on the `action` map

Reply:
Added an assert in the getParameterIndex function. Rather than creating a whole function, I made a variable called paramName that performs the following line in the dispatch function.    
`paramName = key.upToFirstOccurrenceOf(" ", false, false);`
I then use paramName for the lookup table and the key remains unchanged when getParameterIndex is needed.